### PR TITLE
Update Maintainers Guide to current practice

### PR DESCRIPTION
**Pull request by leouieda**
**Description**

Include instructions for using the release checklist issue template
(replaces the instructions here) and include the permissions needed
to make releases.

**Relevant issues/PRs**

### CODE REVIEW

**Comment by leouieda** [comment]
@MGomezN having recently used this guide, would you be able to take a look at these changes and see if they help? Any edits are always welcome!

**Review by MGomezN — APPROVED** [review]
> This page contains instructions for project maintainers about how our setup works, making releases, creating packages, etc.
> 
> Contents
> [Branches](https://github.com/fatiando/community/blob/28d9e758f311a3a169775c1fb17ec068666f9425/MAINTENANCE.md#branches)
> [Reviewing and merging pull requests](https://github.com/fatiando/community/blob/28d9e758f311a3a169775c1fb17ec068666f9425/MAINTENANCE.md#reviewing-and-merging-pull-requests)
> [Continuous Integration](https://github.com/fatiando/community/blob/28d9e758f311a3a169775c1fb17ec068666f9425/MAINTENANCE.md#continuous-integration)
> [Making a Release](https://github.com/fatiando/community/blob/28d9e758f311a3a169775c1fb17ec068666f9425/MAINTENANCE.md#making-a-release)
> [Citations](https://github.com/fatiando/community/blob/28d9e758f311a3a169775c1fb17ec068666f9425/MAINTENANCE.md#citations)

**Subjective comment.** 
I always look for a structure in the content that follows the description in the first paragraph. So I tried to correlate the set A =(how our setup works, making releases, creating packages) with the set B=(Branches, Reviewing and merging pull requests, Continuous Integration, Making a Release, Citations), but for me is not clear if the category  "how our setup works"? Is it related to the section about branches? And where is the info about "creating packages?" is it implied in the making a releases section?

I think someone at the "maintainer" conscience level will not have a problem with this 😕  

> 1. Create an issue with the "Release checklist" template 
>    (source [here](https://github.com/fatiando/.github/blob/main/.github/ISSUE_TEMPLATE/release-checklist.md)).
> 2. Follow the steps in the checklist.

Maybe this is an excellent place to include the link to the Verde Release Youtube video.

**Comment by leouieda** [comment]
> I always look for a structure in the content that follows the description in the first paragraph.

I love that ❤️ I had never thought about this but it makes total sense. I'll have a go at the paragraph to make it match what we actually have in this document.

> Maybe this is an excellent place to include the link to the Verde Release Youtube video.

👍🏽

File: MAINTENANCE.md [hunk 1]
```diff
@@ -1,34 +1,29 @@
...
-This page contains instructions for project maintainers about how our setup
-works, making releases, creating packages, etc.
+> If you want to get involved in the project, you may want to read the
+> [Contributing Guide](CONTRIBUTING.md) instead.
 
-If you want to make a contribution to the project, see the
-[Contributing Guide](CONTRIBUTING.md) instead.
+**This guide documents our procedures and policies for project maintenance
+tasks**, including managing our GitHub repository branches, pull requests,
+continuous integration, releasing, and instructing users on citation practices.
```
Comment (leouieda) [inline MAINTENANCE.md:8]: @MGomezN what do you think? Is this better?

Comment (MGomezN) [reply MAINTENANCE.md:8]: It is great! :)

**Pull request merged**